Modules/_sqlite/module.c

```diff
@@ -404,6 +405,40 @@ pysqlite_error_name(int rc)
     return NULL;
 }
 
+#if SQLITE_VERSION_NUMBER < 3024000
```

Comment:
You need an opposite condition.

Note also that the tests failed. You need them be passed when build with old Sqlite. Since SQLITE_VERSION_NUMBER is not exposed, you have to rely on the result of `hasattr(_sqlite3, 'SQLITE_KEYWORDS')` to determine the expected behavior.